### ISSUE TITLE
os.path.relpath() in gettext builder

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#765]
This line:

```
#!python

# generate "#: file1:line1\n#: file2:line2 ..."
pofile.write(u"#: &#37;s\n" &#37; "\n#: ".join("&#37;s:&#37;s" &#37;
    (path.relpath(source, self.outdir), line)
    for source, line, _ in positions))
```

in sphinx/builders/gettext.py

Throws an exception on W32:

```
  File "&#37;PYTHONPATH&#37;\lib\ntpath.py", line 512, in relpath
    &#37; (path_prefix, start_prefix))
ValueError: path is on drive &#37;DRIVE1&#37;:, start on drive &#37;DRIVE2&#37;:
```

because `source' is absolute path, and self.outdir is also an absolute path (even if sphinx-build was invoked with relative paths for source and output directories), and they could be (and are, in my case) on different drives.

If being able to express source path as relative to self.outdir is a requirement, then i simply don't know how this is going to be fixed. Otherwise, you can catch ValueError here (sticking to better-to-ask-forgiveness-than-to-ask-permission line) and output source path as-is.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/765
- Originally reported by: [LRN](https://bitbucket.org/LRN)
- Originally created at: 2011-09-17T14:15:29.358

**Comment by shimizukawa** [comment]
_From [LRN](https://bitbucket.org/LRN) on 2011-09-17 12:35:34+00:00_

The reason why this is happening is that sphinx doc refers to a number of messages (for ext.pot and web.pot) that come not from its own documentation rst file, but from sphinx itself, for example:
{{{
%PYTHONPATH%\lib\site-packages\sphinx-1.1predev_20110917-py2.7.egg\sphinx\domains__init__.py:docstring of sphinx.domains.Domain:1
}}}
These references might (and do) point to the drive different from the one where documentation rst files are located

I've made a quick hack (as described in my previous post), and was able to build the .pot files successfully. Looking at them, i see now that these paths are put in comments for translatable messages. I am not familiar enough with gettext toolset to be able to claim that these paths will never be read by any piece of software, and that they are there only for translator's convenience, but from my experience it is so. That makes my hack a viable fix, that could be applied to the tip (sorry, don't have any patches for you, too lazy for that).

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-09-21 17:05:28+00:00_

Fix #765: on Windows, relpath() does not always succeed.

→ <<cset d7d7e0287f0a894a54f3e72de10851b96570d526>>
